examples/list.rs

```diff
@@ -1,24 +1,65 @@
-use std::{
-    error::Error,
-    io,
-    time::{Duration, Instant},
-};
+use std::{error::Error, io};
 
 use crossterm::{
     event::{self, DisableMouseCapture, EnableMouseCapture, Event, KeyCode, KeyEventKind},
     execute,
     terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
 };
-use ratatui::{prelude::*, widgets::*};
+use ratatui::{prelude::*, style::palette::tailwind, widgets::*};
+
+const TODO_HEADER_BG: Color = tailwind::BLUE.c950;
+const NORMAL_ROW_COLOR: Color = tailwind::SLATE.c950;
+const ALT_ROW_COLOR: Color = tailwind::SLATE.c900;
+const SELECTED_STYLE_FG: Color = tailwind::BLUE.c300;
+const TEXT_COLOR: Color = tailwind::SLATE.c200;
+const COMPLETED_TEXT_COLOR: Color = tailwind::GREEN.c600;
 
-struct StatefulList<T> {
+struct TodoItem {
+    todo: String,
+    info: String,
+    status: bool,
+}
+
+impl TodoItem {
+    fn styled_list_item(&self, index: usize) -> ListItem {
+        let bg_color = match index % 2 {
+            0 => NORMAL_ROW_COLOR,
+            _ => ALT_ROW_COLOR,
+        };
+        let (style, todo_str) = match self.status {
+            true => (
+                Style::default().fg(COMPLETED_TEXT_COLOR).bg(bg_color),
+                " ✓ ".to_string() + &self.todo,
+            ),
+            false => (
+                Style::default().fg(TEXT_COLOR).bg(bg_color),
+                " ☐ ".to_string() + &self.todo,
+            ),
+        };
+        ListItem::new(Line::from(todo_str)).style(style)
+    }
+
+    fn from_vec(vec: Vec<(&str, &str, bool)>) -> Vec<Self> {
+        let mut result_vec = vec![];
+        for item in vec {
+            result_vec.push(TodoItem {
+                todo: item.0.to_string(),
+                info: item.1.to_string(),
+                status: item.2,
+            })
+        }
+        result_vec
+    }
+}
+
+struct StatefulList {
     state: ListState,
-    items: Vec<T>,
+    items: Vec<TodoItem>,
     last_selected: Option<usize>,
 }
 
-impl<T> StatefulList<T> {
-    fn with_items(items: Vec<T>) -> StatefulList<T> {
+impl StatefulList {
```

Comment:
> If we add first / last methods and keys that would help too.

I like this idea

> What do you think about making there always be a selection?

I think it is good to have this in the example. This might be worth having for some real-life implementations. However, this makes the example harder to read and understand. If simplicity is more important for these examples I think this selection thing can go, else I can try to simplify them.